src/pytorch_lightning/core/module.py

```diff
@@ -1889,7 +1877,7 @@ def to_torchscript(
         """
         mode = self.training
 
-        self._running_torchscript = True
+        LightningModule._running_torchscript = True
```

Comment:
you should still be able to use `self` here as long as this was declared a classattribute before, no?

If you derive a class from it and only set it for `LightningModule`, is it also set for the derived class?

Reply:
> you should still be able to use self here as long as this was declared a classattribute before, no?

I thought so too, but it does not work. Not sure why

> If you derive a class from it and only set it for LightningModule, is it also set for the derived class?


I believe the modified test covers this

Reply:
Maybe you missed an instance attribute? Because if setattr is invoked, the value of Class.attr and instance.attr would diverge.

Reply:
I don't think I did. It might be caused by the torchscript implementation